File: top-starred-opensource-projects/scripts/explore-data.R
## -----------------------------------------------------------------------------
## code: explore-data.R
## description: explore top open source projects on github
## author: anchu@rta.vn
## date: Sat Jul 15 11:06:26 2017
## -----------------------------------------------------------------------------


library(wordcloud)

github.repos <- read.csv("~/Documents/top-starred-opensource-projects/data/TopStaredRepositories.csv",
                         stringsAsFactors = FALSE)


## -----------------------------------------------------------------------------
## data wrangling
## -----------------------------------------------------------------------------


## top list

head(github.repos, 10)

## bottom list
tail(github.repos, 10)

## which are the most popular language?

pop.lang <- table(github.repos$Language[github.repos$Language != ""]) # don't count empty string
pop.lang <- sort(pop.lang)

par(bg = "gray98", cex.axis = 0.8, mar = c(3.5, 7, 2, 1), family = "Lato")
barplot(pop.lang, horiz = TRUE, las = 1, border = FALSE,
        col = adjustcolor("forestgreen", 0.8))
mtext("number of repos", side = 1, line = 2)
mtext("Most Popular Languages Used", line = 0, font = 2)
grid(NULL, NA, col = "gray98", lwd = 1.5, lty = 1)


## which are the most popular tags?

pop.tags <- strsplit(github.repos$Tags, ",")
pop.tags <- unlist(pop.tags)
pop.tags <- sort(table(pop.tags))

par(family = "Lato")
wordcloud(names(pop.tags), pop.tags, scale = c(8, 0.5), max.words = 100,
          random.order = FALSE)


## distribution of stars

stars.count <- gsub("k", "", github.repos$Number.of.Stars)
stars.count <- as.double(stars.count) * 1000

par(family = "Lato", bg = "gray98", cex.axis = 0.8)
hist(log(stars.count), breaks = 50, border = "gray98", col = "steelblue",
     xlim = c(8, 13))

File: top-starred-opensource-projects/scripts/explore-data.py



## -----------------------------------------------------------------------------
## load modules and data
## -----------------------------------------------------------------------------

import pandas as pd
from collections import Counter
import matplotlib.pyplot as plt

github_repos = pd.read_csv("~/Documents/top-starred-opensource-projects/data/TopStaredRepositories.csv")

## -----------------------------------------------------------------------------
## explore data
## -----------------------------------------------------------------------------

# data size
print(github_repos.shape)

print(github_repos.columns)

# top list
print(github_repos.head())


# bottom list
print(github_repos.tail())

# data overall description
print(github_repos.info())

print(github_repos.describe())


# which are the most popular language?

# pop_lang = github_repos["Language"]
# pop_lang[pd.isnull(pop_lang)] = "Unknown"  # same as R 
# pop_lang = Counter(pop_lang)

# pop_lang = pd.DataFrame.from_dict(pop_lang, orient="index").reset_index()
# pop_lang = pop_lang.rename(columns={"index": "lang", 0: "n"})
# pop_lang = pop_lang.sort_values(["n"], ascending=False)

pop_lang = github_repos["Language"].value_counts(dropna=False)
print(pop_lang)


pop_user = github_repos["Username"].value_counts(dropna=False)
print(pop_user)

# which are the most popular tags?

# pop_tag = github_repos["Tags"]
# pop_tag[pd.isnull(pop_tag)] = "Unknown"

# tag_count = []
# [tag_count.extend(tag.split(",")) for tag in pop_tag]

# tag_count = Counter(tag_count)


